lib/tasks/ymodel.rake

```diff
@@ -61,11 +61,10 @@ namespace :ymodel do
         referenced_records.find do |ref|
           ref['id'] == record[old_foreign_key]
         end
-
-      next unless related.present?
-
+      # Some relations are optional
+      new_foreign_val = related.present? ? related[new_index] : nil
```

Comment:
It seems strange to me that we would want to cast `''` to nil. Is there a good reason to do this in the general case? It muddies up the code and might introduce unexpected behaviour.

Reply:
It looked strange to me to see this in a slide:
```
output_element_key: ''
alt_output_element_key: ''
```
instead of 
```
output_element_key:
alt_output_element_key:
```
When there is no related element.
But I agree it muddies up the code, and all relations will work regardless.

Reply:
I thought you agreed that this should be deleted. Could you still remove the change?